src/sparseml/pytorch/sparsification/pruning/mask_creator.py

```diff
@@ -130,9 +130,24 @@ def create_sparsity_masks(
                 masks.append(tensor.new_ones(tensor.shape))
                 continue
 
+            num_elem = tensor.numel()
+            target_num_mask = round(num_elem * sparsity_target)
             min_val = tensor.min().item()
+
             if threshold.item() > min_val:
-                masks.append((tensor > threshold).type(tensor.type()))
+                threshold_mask = (tensor > threshold).type(tensor.type())
```

Comment:
Looks like a 1 off error here. Threshold is calculated so that all params `< threshold` should be pruned. By Keeping params `> threshold`, we're pruning 1 extra each time. A simple `>=` should fix it, but I'm not sure if there's a risk of this not evaluating properly with floats.
```suggestion
                threshold_mask = (tensor >= threshold).type(tensor.type())
```